pulsar-broker/src/main/java/org/apache/pulsar/broker/transaction/buffer/impl/TopicTransactionBuffer.java

```diff
@@ -206,24 +206,28 @@ private void handleTransactionMessage(TxnID txnId, Position position) {
         ByteBuf commitMarker = Markers.newTxnCommitMarker(-1L, txnID.getMostSigBits(),
                 txnID.getLeastSigBits());
 
-        topic.getManagedLedger().asyncAddEntry(commitMarker, new AsyncCallbacks.AddEntryCallback() {
-            @Override
-            public void addComplete(Position position, ByteBuf entryData, Object ctx) {
-                synchronized (TopicTransactionBuffer.this) {
-                    updateMaxReadPosition(txnID);
-                    handleLowWaterMark(txnID, lowWaterMark);
-                    clearAbortedTransactions();
-                    takeSnapshotByChangeTimes();
+        try {
+            topic.getManagedLedger().asyncAddEntry(commitMarker, new AsyncCallbacks.AddEntryCallback() {
```

Comment:
should we release commitMarker in this thread or inside the callback ?

Reply:
in this thread.